Mathlib/Logic/Basic.lean

```diff
@@ -563,6 +567,9 @@ theorem cast_eq_iff_heq : cast e a = a' ↔ HEq a a' :=
   ⟨heq_of_cast_eq _, fun h ↦ by cases h; rfl⟩
 #align cast_eq_iff_heq cast_eq_iff_heq
 
+theorem Eq.rec_eq_cast {α : Sort _} {P : α → Sort _} {x y : α} (h : x = y) (z : P x) :
+    h ▸ z = cast (congr_arg P h) z := by induction h; rfl
+
```

Comment:
I was under the impression that we already had this somewhere...

Reply:
`exact?` cannot find anything. `apply?` suggests after two tries:
`((cast_eq_iff_heq.mpr) <| heq_of_eq_rec_left h rfl).symm `